### PR TITLE
feat(TooltipTrigger): Add onDisplay hook prop

### PR DESCRIPTION
**Pull request by willcumbie**
Add a prop "onDisplay" which will be fired when the tooltip is shown, which works as a seam for adding tracking to tooltips.

### CODE REVIEW

**Comment by adampickeral** [comment]
looks fantastic to me

File: test/TooltipTriggerSpec.jsx [hunk 1]
```diff
@@ -4,14 +4,15 @@ import TooltipTrigger from '../transpiled/TooltipTrigger';
...
+  let trigger, tether, triggerParent, onDisplay;
 
   const renderTrigger = (placement = 'bottom-right') => {
+    onDisplay = jasmine.createSpy('onDisplay');
     tether = jasmine.createSpyObj('tether', ['destroy']);
     spyOn(TooltipTrigger.prototype, '_createTether').and.returnValue(tether);
 
     trigger = TestUtils.renderIntoDocument(
-      <TooltipTrigger placement={placement} content={'The tooltip content'}>
+      <TooltipTrigger placement={placement} content={'The tooltip content'} onDisplay={onDisplay}>
```
Comment (wpwood) [inline test/TooltipTriggerSpec.jsx]: Do we need a test for when onDisplay is not supplied? I could go either way on that one.

**Comment by hannahstwinter** [comment]
LGTM!

File: test/TooltipTriggerSpec.jsx (hunk 1 above)
Comment (willcumbie) [inline test/TooltipTriggerSpec.jsx]: You're right, that was an oversight on my part. Added.

**Pull request merged**